src/math/sqrt.rs

```diff
@@ -95,128 +92,150 @@ pub fn sqrt(x: f64) -> f64 {
             }
         }
     }
-    let mut z: f64;
-    let sign: Wrapping<u32> = Wrapping(0x80000000);
-    let mut ix0: i32;
-    let mut s0: i32;
-    let mut q: i32;
-    let mut m: i32;
-    let mut t: i32;
-    let mut i: i32;
-    let mut r: Wrapping<u32>;
-    let mut t1: Wrapping<u32>;
-    let mut s1: Wrapping<u32>;
-    let mut ix1: Wrapping<u32>;
-    let mut q1: Wrapping<u32>;
+    #[cfg(target_feature = "sse2")]
+    {
+        // Note(Lokathor): If compile time settings allow, we just use SSE2, since
+        // the sqrt in `std` on these platforms also compiles down to an SSE2
+        // instruction.
+        #[cfg(target_arch = "x86")]
+        use core::arch::x86::*;
+        #[cfg(target_arch = "x86_64")]
+        use core::arch::x86_64::*;
+        unsafe {
+            let m = _mm_set_sd(x);
+            let m_sqrt = _mm_sqrt_pd(m);
+            _mm_cvtsd_f64(m_sqrt)
+        }
```

Comment:
Doesn't using the same code here as for wasm32 produce the exact same effect for targets with SSE2? If not, why not ?

Reply:
I think I don't understand the question. `sse2` is an x86 and x86_64 feature. It will never be available during a wasm32 build.

Reply:
I meant that instead of using the `core::arch` intrinsics here, you can just write `unsafe { ::core::intrinsics::sqrtf64(x) }` just like wasm32 does, and that should generate this exact same code if SSE2 is enabled.

Reply:
For example, see: https://rust.godbolt.org/z/eEF9ef

Reply:
ah, but that's not stable. this code is. that's why i did not also add a neon version, I'm only out to help out stable rust today.

you're right that we should probably also reconsider  calling the core intrinsic in more cases if Nightly is in use.